src/foam/core/XMLSupport.java

```diff
@@ -0,0 +1,158 @@
+            PropertyInfo prop = null;
+            while (reader.hasNext()) {
+                eventType = reader.next();
+                switch (eventType){
+                    case XMLStreamConstants.START_ELEMENT:
+                        prop = (PropertyInfo) obj.getClassInfo().getAxiomByName(reader.getLocalName());
+                        if (prop == null) {
+                            throw new XMLStreamException("Could not find property " + reader.getLocalName());
+                        }
+                        break;
+                    case XMLStreamConstants.CHARACTERS:
+                        if (prop != null) {
+                            prop.set(obj, reader.getText());
+                            prop = null;
+                        }
+                        break;
+                    case XMLStreamConstants.END_ELEMENT:
+                        if(reader.getLocalName().equals("object")) { return; }
+                        break;
+                }
+            }
+        } catch (XMLStreamException ex ) {
+            throw new XMLStreamException("Premature end of xml file");
+        }
+    }
+
+    // FObjects taken from XMLDAO and written to file with specified filename given
+    public static void FObjectToXML (String fileName, XMLDAO xmlDAO) throws XMLStreamException {
+        // Returns all objects into a list iterator
+        ListSink ls = new ListSink();
+        xmlDAO.select(ls);
+        List objList = ls.getData();
+        Iterator objItr = objList.iterator();
+
+        try {
+            XMLOutputFactory xMLOutputFactory = XMLOutputFactory.newInstance();
+            XMLStreamWriter xmlStreamWriter = xMLOutputFactory.createXMLStreamWriter(new FileWriter(fileName));
+            xmlStreamWriter = new IndentingXMLStreamWriter(xmlStreamWriter);
+
+            xmlStreamWriter.writeStartDocument();
+            //Root tag
+            xmlStreamWriter.writeStartElement("objects");
+
+            while (objItr.hasNext()) {
+                FObject currentObj = (FObject)objItr.next();
+                xmlStreamWriter.writeStartElement("object");
+                xmlStreamWriter.writeAttribute("class", currentObj.getClass().toString().replaceAll("class ",""));
+
+                List props = currentObj.getClassInfo().getAxioms();
+                Iterator propItr = props.iterator();
+
+                while (propItr.hasNext()) {
+                    PropertyInfo currentProp = (PropertyInfo)propItr.next();
+                    Object value = currentProp.f(currentObj);
+                    if (value != null && value != "" ) {
+                        xmlStreamWriter.writeStartElement(currentProp.getName());
+                        // Case for date
+                        if (value instanceof java.util.Date) {
+                            String dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+                            LocalDateTime ldt = LocalDateTime.ofInstant(((Date) value).toInstant(), ZoneId.of("UTC"));
```

Comment:
I don't think date format should be local specific. If we use this to store unit tests for FOAM/NANOS, someone will checkout the code is some different local and then they won't be able to read the data.